pkg/http/handler/timeout_test.go

```diff
@@ -24,15 +25,68 @@ import (
 	"time"
 )
 
+func TestTimeoutWriterAllowsForAdditionalWrites(t *testing.T) {
+	recorder := httptest.NewRecorder()
+	handler := &timeoutWriter{w: recorder}
+	handler.WriteHeader(http.StatusOK)
+	handler.TimeoutAndWriteError("error")
+	if _, err := io.WriteString(handler, "test"); err != nil {
+		t.Fatalf("handler.Write() = %v, want no error", err)
+	}
+
+	if got, want := recorder.Code, http.StatusOK; got != want {
+		t.Errorf("recorder.Status = %d, want %d", got, want)
+	}
+	if got, want := recorder.Body.String(), "test"; got != want {
+		t.Errorf("recorder.Body = %s, want %s", got, want)
+	}
+}
+
+func TestTimeoutWriterDoesntFlushAfterTimeout(t *testing.T) {
+	recorder := httptest.NewRecorder()
+	handler := &timeoutWriter{w: recorder}
+	handler.TimeoutAndWriteError("error")
+	handler.Flush()
+
+	if got, want := recorder.Flushed, false; got != want {
+		t.Errorf("recorder.Flushed = %t, want %t", got, want)
+	}
+}
+
+func TestTimeoutWriterFlushesAfterTimeout(t *testing.T) {
+	recorder := httptest.NewRecorder()
+	handler := &timeoutWriter{w: recorder}
+	handler.Flush()
+
+	if got, want := recorder.Flushed, true; got != want {
+		t.Errorf("recorder.Flushed = %t, want %t", got, want)
+	}
+}
+
+func TestTimeoutWriterErrorsWriteAfterTimeout(t *testing.T) {
+	recorder := httptest.NewRecorder()
+	handler := &timeoutWriter{w: recorder}
+	handler.TimeoutAndWriteError("error")
+	if _, err := handler.Write([]byte("hello")); err != http.ErrHandlerTimeout {
+		t.Errorf("ErrHandlerTimeout got %v, want: %s", err, http.ErrHandlerTimeout)
+	}
+}
+
 func TestTimeToFirstByteTimeoutHandler(t *testing.T) {
 	const (
-		failingTimeout = 0 * time.Millisecond
-		longTimeout    = 10 * time.Second
+		longTimeout = 10 * time.Second
```

Comment:
is this one still used elsewhere?

Reply:
Yes. It is used in the table tests later.